pydantic/main.py

```diff
@@ -1216,6 +1237,49 @@ def _calculate_keys(self, *args: Any, **kwargs: Any) -> Any:
         return _deprecated_copy_internals._calculate_keys(self, *args, **kwargs)  # type: ignore
 
 
+RootModelRootType = typing.TypeVar('RootModelRootType')
+
+
+class RootModel(BaseModel, typing.Generic[RootModelRootType]):
+    __pydantic_root_model__ = True
+
+    root: RootModelRootType
+
+    def __init__(__pydantic_self__, data: RootModelRootType) -> None:  # type: ignore
+        __tracebackhide__ = True
+        __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
+
+    __init__.__pydantic_base_init__ = True  # type: ignore
+
+    def model_dump(
+        self,
+        *,
+        mode: str = 'python',
+        include: IncEx = None,
+        exclude: IncEx = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> dict[str, Any]:
+        return super().model_dump(
+            mode=mode,
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            round_trip=round_trip,
+            warnings=warnings,
+        )['root']
```

Comment:
@samuelcolvin Shall we return an actual value from the pydantic-core part instead of a dict with a single attribute? It seems logical as we require a bare value for initializing the schema.

Reply:
It has been set. I'll add a check for that in the test.

Reply:
https://github.com/pydantic/pydantic/pull/5765/files#diff-41d180f6f7233d175c2dfe36c45c4ea80eed4754fa7ed4eed46ccde1b2a778c7R2270

`root_model is True`

Reply:
@samuelcolvin I don't see anything in `pydantic-core/src/serializers` that references `root_model` so I think there probably needs to be some tweaks for it to extract that field automatically

Reply:
@lig I do think we should get that fixed in pydantic-core before merging this because I think without a fix there, we won't get proper serialization when using a RootModel as a field

Reply:
Totally agree

Reply:
Fixed in https://github.com/pydantic/pydantic-core/pull/613